### PR TITLE
fix: dataset data-types count loading

### PR DESCRIPTION
**Pull request by v-rocheleau**
Small fix for dataset data-types count stuck on loading.
The dataset's data-types were not properly accessed by ID.

### CODE REVIEW

**Review by noctillion — APPROVED** [review]
lgtm

**Review by gsfk — APPROVED** [review]
Looks good. 

The actual rendering of this value is a bit weird, but that's an issue elsewhere: it flashes weirdly for me in both Chrome and Firefox: 

flashy_three

.. but this can be handled elsewhere

File: src/components/datasets/DatasetOverview.js [hunk 1]
```diff
@@ -4,14 +4,15 @@ import PropTypes from "prop-types";
...
-import { useDatasetDataTypes } from "@/modules/datasets/hooks";
+import { useDatsetDataTypeById } from "@/modules/datasets/hooks";
 import { datasetPropTypesShape, projectPropTypesShape } from "@/propTypes";
 
 const DatasetOverview = ({ isPrivate, project, dataset }) => {
-  const datasetsDataTypes = useDatasetDataTypes();
-  const datasetDataTypes = datasetsDataTypes[dataset.identifier];
+  // const datasetsDataTypes = useDatasetDataTypes();
+  // const datasetDataTypes = datasetsDataTypes.itemsByID[dataset.identifier];
+  const datasetDataTypes = useDatsetDataTypeById(dataset.identifier);
```
Comment (davidlougheed) [inline src/components/datasets/DatasetOverview.js:13]: typo: datset

File: src/components/datasets/DatasetOverview.js [hunk 2]
```diff
@@ -4,14 +4,15 @@ import PropTypes from "prop-types";
...
   const isFetchingDataset = datasetDataTypes?.isFetching;
-
+  console.log(isFetchingDataset);
```
Comment (davidlougheed) [inline src/components/datasets/DatasetOverview.js:15]: rm after done debugging

File: src/modules/datasets/hooks.js [hunk 3]
```diff
@@ -13,3 +17,15 @@ export const useDatasetDataTypes = () => {
   }, [dispatch, datasetsByID]);
   return useAppSelector((state) => state.datasetDataTypes);
 };
+
+export const useDatsetDataTypeById = (datasetId) => {
```
Comment (davidlougheed) [inline src/modules/datasets/hooks.js:21]: should be `useDatasetDataTypesById` (or `ID` depending on what other hooks use)

**Review by davidlougheed — CHANGES_REQUESTED** [review]

File: src/modules/datasets/hooks.js [hunk 4]
```diff
@@ -13,3 +17,15 @@ export const useDatasetDataTypes = () => {
   }, [dispatch, datasetsByID]);
   return useAppSelector((state) => state.datasetDataTypes);
 };
+
+export const useDatasetDataTypeByID = (datasetId) => {
```
Comment (davidlougheed) [inline src/modules/datasets/hooks.js:21]: should be plural I think - multiple data types?

**Review by davidlougheed — CHANGES_REQUESTED** [review]

File: src/modules/datasets/hooks.js (hunk 4 above)
Comment (v-rocheleau) [reply src/modules/datasets/hooks.js:21]: Oh true

File: src/components/datasets/DatasetOverview.js [hunk 5]
```diff
@@ -4,14 +4,12 @@ import PropTypes from "prop-types";
...
+import { useDatasetDataTypesByID } from "@/modules/datasets/hooks";
 import { datasetPropTypesShape, projectPropTypesShape } from "@/propTypes";
 
 const DatasetOverview = ({ isPrivate, project, dataset }) => {
-  const datasetsDataTypes = useDatasetDataTypes();
-  const datasetDataTypes = datasetsDataTypes[dataset.identifier];
+  const datasetDataTypes = useDatasetDataTypesByID(dataset.identifier);
   const isFetchingDataset = datasetDataTypes?.isFetching;
-
   // Count data types which actually have data in them for showing in the overview
```
Comment (davidlougheed) [inline src/components/datasets/DatasetOverview.js:13]: we could use `isFetching` here from `datasetDataTypes` as another loading condition for the data types statistic

**Review by davidlougheed — APPROVED** [review]
overall looks good, but I think we could do more with the `isFetching` inside `useDatasetDataTypesByID` to make loading logic more complete for the data type count statistic.

File: src/components/datasets/DatasetOverview.js (hunk 5 above)
Comment (v-rocheleau) [reply src/components/datasets/DatasetOverview.js:13]: I added `hasAttempted` in this store to only show the dash when waiting on the first fetch.

Comment (davidlougheed) [reply src/components/datasets/DatasetOverview.js:13]: nice!

**Review by davidlougheed — APPROVED** [review]
lgtm, nice!

**Pull request merged**